### PR TITLE
Updating MultiTask example to use new infer api and adding test for CI

### PR DESCRIPTION
**Pull request by andrewfayres**
## Description ##
Updated the multitask example to use the new inference api. Also added a test case that will run the example.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+  // TODO: Need to confirm with Windows
+
+
+  Process("unzip " + tempDirPath + "/multitask/mnist.zip -d "
+    + tempDirPath + "/multitask/") !
+
+  val batchSize = 100
+  val numEpoch = 10
+  val ctx = Context.cpu()
```
Comment (marcoabreu) [inline scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: @nswamy is there an auto context in Scala?

**Comment by marcoabreu** [comment]
Thanks a lot for adding the examples to our CI. In general, I really like the idea and would like to strive for having all examples in CI. I'm just a bit concerned that the training time in our examples could cause some problems, thus, in the long term I'd like to move them into a separate job.

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  val lr = 0.001f
+  val network = ExampleMultiTask.buildNetwork()
+  val (trainIter, valIter) =
+    Data.mnistIterator(modelDirPath, batchSize = batchSize, inputShape = Shape(784))
+  val trainMultiIter = new ExampleMultiTask.MultiMnistIterator(trainIter)
+  val valMultiIter = new ExampleMultiTask.MultiMnistIterator(valIter)
+
+  val datasAndLabels = trainMultiIter.provideData ++ trainMultiIter.provideLabel
+
+  val (argShapes, outputShapes, auxShapes) = network.inferShape(trainMultiIter.provideData("data"))
```
Comment (nswamy) [inline scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: bulk of the code in this file is training code, can you move it to generic runTraining method inside of the example, so customers can directly use it?  you can call the runTraining code by passing training/test files, etc.,

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala (hunk 1 above)
Comment (nswamy) [reply scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: we don't have auto context, we planned to run these examples only on GPUs since it takes more time on CPU, We currently control this using a environment variable that is passed from a script. We will be moving away from this model with #11596

Comment (yzhliu) [reply scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: what is auto context?

**Comment by yzhliu** [comment]
Can we move test code to `test` package and name it XXXSuite` to follow the Scala test convention?

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala (hunk 1 above)
Comment (marcoabreu) [reply scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: Auto context is a method we use in our Python frontend where the context gets globally defined. The script then takes this configuration and uses it instead having the context explicitly defined.

Otherwise you run into the problem that you are not able to test the same code on CPU and GPU without having to make a Switch somewhere.

Thanks for elaborating, Naveen. I agree that GPU would make sense here. At the moment, it states CPU - is that a mistake or on purpose?

Comment (andrewfayres) [reply scala-package/examples/src/test/scala/org/apache/mxnetexamples/multitask/MultiTaskTest.scala]: I did CPU on purpose because this is training on MNIST data. It trains on my macbook in just a few minutes.

This is consistent with the other example that we have that uses mnist [here](https://github.com/apache/incubator-mxnet/blob/master/scala-package/examples/src/test/scala/org/apache/mxnetexamples/imclassification/MNISTExampleSuite.scala#L53).
